src/TodoContext/TodoContext.tsx

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React, { Dispatch, SetStateAction, useCallback, useState } from 'react';
+import { ErrorMessages, Todo } from '../types/Todo';
+import * as todoService from '../../src/api/todos';
+
+type ContextType = {
+  error: ErrorMessages | null;
+  tempTodo: Todo | null;
+  setTempTodo: (todo: Todo | null) => void;
+  setError: (value: ErrorMessages | null) => void;
+  displayError: (message: ErrorMessages) => void;
+  isLoading: boolean;
+  setIsLoading: (value: boolean) => void;
+  todos: Todo[];
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+  activeTodos: Todo[];
+  setActiveTodos: (todos: Todo[]) => void;
+  handleDeleteTodo: (todo: Todo) => void;
+  toggleTodo: (todo: Todo) => void;
+  renameTodo: (todo: Todo, newTitle: string) => Promise<void>;
+};
+
+export const TodoContext = React.createContext<ContextType>({
+  error: null,
+  setError: (_value: ErrorMessages | null) => {},
+  displayError: (_message: ErrorMessages) => {},
+  tempTodo: null,
+  setTempTodo: (_todo: Todo | null) => {},
+  isLoading: false,
+  setIsLoading: (_value: boolean) => {},
+  todos: [],
+  setTodos: () => [],
+  activeTodos: [],
+  setActiveTodos: (_todos: Todo[]) => {},
+  handleDeleteTodo: (_todo: Todo) => {},
+  toggleTodo: (_todo: Todo) => {},
+  renameTodo: (_todo: Todo, _newTitle: string) => new Promise(() => {}),
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [error, setError] = useState<ErrorMessages | null>(null);
```

Comment:
Create a default option in your `ErrorMessages` enum, instead of using `null` here

Reply:
Do you want me to do `useState<ErrorMessages>(ErrorMessages.Default)`?
But I'm using `null` value to show error component or hide.

Reply:
> > Do you want me to do `useState<ErrorMessages>(ErrorMessages.Default)`? But I'm using `null` value to hide error component.
> 
> `ErrorMessages.Default` could be an empty string and you can hide it in the same way. But in my opinion null is ok too

🤷